pkg/controllers/provisioning/provisioner.go

```diff
@@ -308,11 +308,14 @@ func (p *Provisioner) Schedule(ctx context.Context) (*scheduler.Results, error)
 	if len(pods) == 0 {
 		return &scheduler.Results{}, nil
 	}
-	scheduler, err := p.NewScheduler(ctx, pods, nodes.Active(), scheduler.SchedulerOptions{})
+	s, err := p.NewScheduler(ctx, pods, nodes.Active(), scheduler.SchedulerOptions{})
 	if err != nil {
 		return nil, fmt.Errorf("creating scheduler, %w", err)
 	}
-	return scheduler.Solve(ctx, pods)
+	if s == nil {
```

Comment:
Can we return a nil scheduler? This should just be an invariant in our code we enforce